File: SnowForecast/app/src/main/java/com/sners/snowforecast/view/MinutelyPrecipChart.java
package com.sners.snowforecast.view;

import java.util.ArrayList;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.view.animation.Animation;
import android.view.animation.AnimationUtils;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import com.sners.snowforecast.ForecastMainActivity;
import com.sners.snowforecast.R;
import com.sners.snowforecast.data.IntervalData;
import com.sners.snowforecast.view.HourlyPrecipChart;


public class MinutelyPrecipChart extends com.sners.snowforecast.view.WeatherPrecipChart {

	final int numPointsToPlot = 60;
	
	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		
	    setContentView(R.layout.precip_chart);
	    ivPcIntensity	= (ImageView)findViewById(R.id.ivPcIntensity);
	    ivPcProbability	= (ImageView)findViewById(R.id.ivPcProbability);
	    tvPcIntensity	= (TextView) findViewById(R.id.tvPcIntensity);
	    
	    precipPrefix = "Minutely Precip: ";
	    
        ArrayList<IntervalData> minutely = ForecastMainActivity.weatherData.getMinutelyData();
        Float maxPrecip = ForecastMainActivity.weatherData.getMinutely().getMaxPrecip();
        
        maxPrecip = setTitleAndColours(maxPrecip);
    
	    drawPrecipGraph(minutely, maxPrecip, numPointsToPlot);

        drawProbabilityGraph(minutely, numPointsToPlot);
	}

	public void displayDashboard(View v){
		//Animation animExitLeft = AnimationUtils.makeOutAnimation(this, true);
		//llCurrently.startAnimation(animExitLeft);
		Intent nextActivityIntent;
		String toastMessage;
		if (null != ForecastMainActivity.weatherData.getHourly() && ForecastMainActivity.weatherData.getHourly().getMaxPrecip() > 0.0f)
		{
			nextActivityIntent = new Intent(MinutelyPrecipChart.this, HourlyPrecipChart.class);
			toastMessage = "Show Hourly graphs";		}
		else
		{
			nextActivityIntent = new Intent(MinutelyPrecipChart.this, WeatherDashboard.class);
			toastMessage = "Show the Dashboard";
		}
		Animation animExitLeft = AnimationUtils.makeOutAnimation(this, true);
		Toast.makeText(getApplicationContext(), toastMessage, 
				Toast.LENGTH_SHORT).show();
		startActivity(nextActivityIntent);

		finish();

	}

}

File: SnowForecast/app/src/main/java/com/sners/snowforecast/view/MinutelyPrecipChart.kt
package com.sners.snowforecast.view

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.view.animation.AnimationUtils
import android.widget.ImageView
import android.widget.TextView
import android.widget.Toast
import com.sners.snowforecast.ForecastMainActivity
import com.sners.snowforecast.R
import com.sners.snowforecast.weather.WeatherConstants

/**
 * A class representing the Minutely graphs view
 */
class MinutelyPrecipChart : WeatherPrecipChart() {

    /**
     * On Create - Override
     * @param savedInstanceState Saved instance state
     */
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.precip_chart)
        ivPcIntensity = findViewById<View>(R.id.ivPcIntensity) as ImageView
        ivPcProbability = findViewById<View>(R.id.ivPcProbability) as ImageView
        tvPcIntensity = findViewById<View>(R.id.tvPcIntensity) as TextView
        precipPrefix = "Minutely Precip: "
        val minutely = ForecastMainActivity.weatherData.minutelyData
        var maxPrecip: Float = ForecastMainActivity.weatherData.minutely!!.getMaxPrecip()
        maxPrecip = setTitleAndColours(maxPrecip)
        drawPrecipGraph(minutely, maxPrecip, WeatherConstants.MINUTELY_NUM_POINTS_TO_PLOT)
        drawProbabilityGraph(minutely, WeatherConstants.MINUTELY_NUM_POINTS_TO_PLOT)
    }

    /**
     * Display Current Weather
     * @param v View representing the current view - comes in from the layout
     */
    fun displayDashboard(v: View?) {
        //Animation animExitLeft = AnimationUtils.makeOutAnimation(this, true);
        //llCurrently.startAnimation(animExitLeft);
        val nextActivityIntent: Intent
        val toastMessage: String
        if (null != ForecastMainActivity.weatherData.hourly && ForecastMainActivity.weatherData.hourly!!.getMaxPrecip() > 0.0f) {
            nextActivityIntent = Intent(this@MinutelyPrecipChart, HourlyPrecipChart::class.java)
            toastMessage = "Show Hourly graphs"
        } else {
            nextActivityIntent = Intent(this@MinutelyPrecipChart, WeatherDashboard::class.java)
            toastMessage = "Show the Dashboard"
        }
        val animExitLeft = AnimationUtils.makeOutAnimation(this, true)
        Toast.makeText(
            applicationContext, toastMessage,
            Toast.LENGTH_SHORT
        ).show()
        startActivity(nextActivityIntent)
        finish()
    }
}